src/eve/visitors.py

```diff
@@ -118,6 +128,10 @@ def visit(self, node: concepts.TreeNode, **kwargs: Any) -> Any:
                 if node_class is concepts.Node:
                     break
 
+        if self.previsitors:
+            for previsitor in self.previsitors:
+                kwargs = previsitor(node, kwargs)
```

Comment:
As mentioned above, I think it is more coherent to pass the unpacked `**kwargs` here:
```suggestion
                kwargs = previsitor(node, **kwargs)
```

Reply:
Since we pass both the `NodeVisitor` and the `kwargs` now, I think I'll skip returning both and just pass the dict so they can be updated in-place.